functions.lua

```diff
@@ -362,6 +433,131 @@ function M.get_crossref(str)
+  elseif str:find('\\[eE][pP][rR][iI][nN][tT]%s*%{%s*[^%s%}]+%d%s*%}') then
+    arxiv = str:match('\\[eE][pP][rR][iI][nN][tT]%s*%{%s*([^%s%}]+%d)%s*%}')
+  end
+  if arxiv~='' or str_naked:match('[Pp]?[rR?][eE]%-?[pP][rR][iI][nN][tT]') or str_naked:match('[aA][rR][xX][iI][vV]') then -- or str_naked:match('[Pp][rR][eE][pP][aA][rR][aA][tT][iI][oO][nN]')
+  -- if arXiv ID found OR the entry is a preprint or arXiv...
+    local function isTable(t) return t and type(t) == 'table' end
+    if arxiv~='' then
+     -- arXiv number found, querying arXiv API for arXiv ID
+     axret = M.execute('arXiv found: '..arxiv..', checking arXiv API... ' --..i..' of '..#bibitems
+      , false,
+      'wget -qO-', M.quote(C.arxivapi .. M.escapeUrl(arxiv)))
+    else
+     -- arXiv number NOT found, querying arXiv API for (naked) bibitem value for best match...
+-- print (C.arxivapiq .. M.escapeUrl(str_naked:gsub("^%s*(.-)%s*$", "%1")))
+     axret = M.execute('  querying arXiv for '..str_naked:gsub("^%s*(.-)%s*$", "%1"), --..i..' of '..#bibitems
+      false,
+      'wget -qO-', M.quote(C.arxivapiq .. M.escapeUrl(str_naked:gsub("^%s*(.-)%s*$", "%1"))))
+    end
+    local xmlhandler = require("xmlhtree")
+    local axHandler = xmlhandler:new()
+    local xmlparser = xml2lua.parser(axHandler)
+--  print (axret)
+    xmlparser:parse(axret)
+    local axentry = axHandler and axHandler.root and type(axHandler.root)=='table' and axHandler.root.feed and type(axHandler.root.feed)=='table' and axHandler.root.feed.entry
+--[[
+If there is more than one person, then person is an array instead of a regular table.
+This way, we need to iterate over the person array instead of the people table.
+]]
+-- if type(axentry)=='table' then M.printTable(axentry,1) end
+    local axauthors=''
+    if axentry and axentry.author then
+      if type(axentry.author)=='table' then
+        if axentry.author.name then
+          axauthors=axentry.author.name
+        else
+         for axauk, axauv in ipairs(axentry.author) do
+          if (axauk > 1) then axauthors=axauthors..' and ' end
+            axauthors=axauthors..axauv.name
+         end
+        end
+      else
+        axauthors=axentry.author
+      end
+--      print ('FOUND AX authors: '..axauthors)
+      ret = ret..'      AUTHOR = {'..axauthors..'},\n';
+--      if arxiv:find('v%d+$') then
+--        ret = ret..'      YEAR = {'..string.sub(axentry.updated,1,4)..'},\n';
+--      elseif axentry.id:match('v(%d+)$')=='1' or 
+      if string.sub(axentry.published,1,4)==string.sub(axentry.updated,1,4) then
+        ret = ret..'      YEAR = {'..string.sub(axentry.published,1,4)..'},\n';
+      else
+        ret = ret..'      YEAR = {[v1]~'..string.sub(axentry.published,1,4)..', [v'..axentry.id:match('v(%d+)$')..']~'..string.sub(axentry.updated,1,4)..'},\n';
+      end
+      ret = ret..'      EPRINT = {'..axentry.id..'},\n';
+--      ret = ret..'      EPRINT = {'..axentry.id..'},\n';
```

Comment:
is there a reason these comments are left in here?

Reply:
Thanks, I removed these unneeded comments of previous code versions.

Reply:
(see the changes of the latest commit, also the bibcheck.lua, of the PR)